### PR TITLE
feature: add os type FreeBSD

### PR DESCRIPTION
**Pull request by penggehero**
### 新加的功能:
- 操作系统类型枚举字段增加“FreeBSD”枚举项

### CODE REVIEW

**Comment by tencentblueking-adm** [comment]
[CLA assistant check](https://bk-cla.bktencent.com:443/TencentBlueKing/bk-cmdb?pullRequest=6667) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bk-cla.bktencent.com:443/TencentBlueKing/bk-cmdb?pullRequest=6667) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bk-cla.bktencent.com:443/check/TencentBlueKing/bk-cmdb?pullRequest=6667) it.</sub>

File: src/scene_server/admin_server/upgrader/y3.10.202303072103/add_freebsd_property.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
+// Package y3_10_202303072103 TODO
```
Comment (wjyfelen) [inline src/scene_server/admin_server/upgrader/y3.10.202303072103/add_freebsd_property.go:1]: 文件头不是这样的吧？

File: src/scene_server/admin_server/upgrader/y3.10.202303072103/add_freebsd_property.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+import (
+	"configcenter/src/common"
+	"configcenter/src/common/condition"
+	"configcenter/src/common/mapstr"
+	"configcenter/src/common/metadata"
+	"configcenter/src/scene_server/admin_server/upgrader"
+	"configcenter/src/storage/dal"
+	"context"
+	"time"
```
Comment (wjyfelen) [inline src/scene_server/admin_server/upgrader/y3.10.202303072103/add_freebsd_property.go:28]: import包的顺序

File: src/scene_server/admin_server/upgrader/y3.10.202303072103/add_freebsd_property.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	IsAPI             bool        `json:"bk_isapi" bson:"bk_isapi"`
+	PropertyType      string      `json:"bk_property_type" bson:"bk_property_type"`
+	Option            interface{} `json:"option" bson:"option"`
+	Description       string      `json:"description" bson:"description"`
+	Creator           string      `json:"creator" bson:"creator"`
+	CreateTime        *time.Time  `json:"create_time" bson:"create_time"`
+	LastTime          *time.Time  `json:"last_time" bson:"last_time"`
+}
+
+func addFreeBSDProperty(ctx context.Context, db dal.RDB, conf *upgrader.Config) error {
```
Comment (wjyfelen) [inline src/scene_server/admin_server/upgrader/y3.10.202303072103/add_freebsd_property.go:57]: conf 没有用到就不用传了

File: src/scene_server/admin_server/upgrader/y3.10.202303072103/pkg.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+func init() {
+	upgrader.RegistUpgrader("y3.10.202303072103", upgrade)
+}
+
+func upgrade(ctx context.Context, db dal.RDB, conf *upgrader.Config) (err error) {
+	blog.Infof("start execute y3.10.202303072103")
+
+	if err = addFreeBSDProperty(ctx, db); err != nil {
+		blog.Errorf("upgrade y3.10.202303072103 add FreeBSD property  failed, err: %v", err)
```
Comment (laodiu) [inline src/scene_server/admin_server/upgrader/y3.10.202303072103/pkg.go:36]: property后面多了个空格

File: src/scene_server/datacollection/collections/hostsnap/hostsnap.go [hunk 5]
```diff
@@ -411,6 +411,9 @@ func parseSetter(val *gjson.Result, innerIP, outerIP string) (map[string]interfa
 	case "hp-ux":
 		osname = platform
 		ostype = common.HostOSTypeEnumHpUX
+	case "freebsd":
```
Comment (laodiu) [inline src/scene_server/datacollection/collections/hostsnap/hostsnap.go:414]: freebsd这个字段有和采集器那边确认好吗？

File: src/scene_server/admin_server/upgrader/y3.10.202303072103/pkg.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	"configcenter/src/common/blog"
+	"configcenter/src/scene_server/admin_server/upgrader"
+	"configcenter/src/storage/dal"
+)
+
+func init() {
+	upgrader.RegistUpgrader("y3.10.202303072103", upgrade)
+}
+
+func upgrade(ctx context.Context, db dal.RDB, conf *upgrader.Config) (err error) {
```
Comment (laodiu) [inline src/scene_server/admin_server/upgrader/y3.10.202303072103/pkg.go:32]: conf参数没用到，删了吧

Comment (laodiu) [inline src/scene_server/admin_server/upgrader/y3.10.202303072103/pkg.go:32]: conf参数没用到，删了吧

File: src/scene_server/admin_server/upgrader/y3.10.202303072103/add_freebsd_property.go [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+import (
+	"context"
+	"time"
+
+	"configcenter/src/common"
+	"configcenter/src/common/metadata"
+	"configcenter/src/storage/dal"
+)
+
+type attribute struct {
```
Comment (ZQHcode) [inline src/scene_server/admin_server/upgrader/y3.10.202303072103/add_freebsd_property.go:29]: 这个结构体不是最新的，至少缺了IsMultiple字段，可以和最新的upgrader对比下

File: src/scene_server/admin_server/upgrader/y3.10.202303072103/add_freebsd_property.go [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+	IsAPI             bool           `field:"bk_isapi" json:"bk_isapi" bson:"bk_isapi" mapstructure:"bk_isapi"`
+	PropertyType      string         `field:"bk_property_type" json:"bk_property_type" bson:"bk_property_type" mapstructure:"bk_property_type"`
+	Option            interface{}    `field:"option" json:"option" bson:"option" mapstructure:"option"`
+	IsMultiple        *bool          `field:"ismultiple" json:"ismultiple,omitempty" bson:"ismultiple" mapstructure:"ismultiple"`
+	Description       string         `field:"description" json:"description" bson:"description" mapstructure:"description"`
+	Creator           string         `field:"creator" json:"creator" bson:"creator" mapstructure:"creator"`
+	CreateTime        *metadata.Time `json:"create_time" bson:"create_time" mapstructure:"create_time"`
+	LastTime          *metadata.Time `json:"last_time" bson:"last_time" mapstructure:"last_time"`
+}
+
```
Comment (laodiu) [inline src/scene_server/admin_server/upgrader/y3.10.202303072103/add_freebsd_property.go:55]: 这里应该用不到mapstructure这个tag，可以去掉

**Review by wjyfelen — APPROVED** [review]

File: src/scene_server/admin_server/upgrader/y3.10.202303072103/add_freebsd_property.go [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+	IsEditable        bool           `field:"editable" json:"editable" bson:"editable"`
+	IsPre             bool           `field:"ispre" json:"ispre" bson:"ispre"`
+	IsRequired        bool           `field:"isrequired" json:"isrequired" bson:"isrequired"`
+	IsReadOnly        bool           `field:"isreadonly" json:"isreadonly" bson:"isreadonly"`
+	IsOnly            bool           `field:"isonly" json:"isonly" bson:"isonly"`
+	IsSystem          bool           `field:"bk_issystem" json:"bk_issystem" bson:"bk_issystem"`
+	IsAPI             bool           `field:"bk_isapi" json:"bk_isapi" bson:"bk_isapi"`
+	PropertyType      string         `field:"bk_property_type" json:"bk_property_type" bson:"bk_property_type"`
+	Option            interface{}    `field:"option" json:"option" bson:"option"`
+	IsMultiple        *bool          `field:"ismultiple" json:"ismultiple,omitempty" bson:"ismultiple"`
```
Comment (ZQHcode) [inline src/scene_server/admin_server/upgrader/y3.10.202303072103/add_freebsd_property.go:49]: 这个是指针类型，确认下如果存的是nil，会不会有问题

**Review by ZQHcode — DISMISSED** [review]

**Review by ZQHcode — APPROVED** [review]

**Pull request merged**